lib/modelManagers/userRoleManager.js

```diff
@@ -0,0 +1,87 @@
+const { Sequelize } = require('sequelize'),
+  { User, Roles, UserRoles } = require('../../database/models');
+
+const { Op } = Sequelize;
+
+/**
+ * @description Model Manager for Article and Admin
+ *
+ * @class UserRoleManager
+ */
+class RoleManager {
+  /**
+   * @description this manager checks admin authentication it check if the user is an admin
+   * @param {*} userId
+   * @static
+   * @returns {*} admin
+   * @memberof RoleManager
+   */
+  static async findUserRole(userId) {
+    const options = {
+      where: {
+        id: {
+          [Op.eq]: userId
+        }
+      },
+      include: [
+        {
+          model: Roles,
+          attributes: ['id', 'role']
+        }
+      ],
+      attributes: ['id', 'firstname', 'lastname', 'email']
+    };
+    const admin = await User.findOne(options);
+    return admin;
+  }
+
+  /**
+   *
+   * @description the manager create admin role if the need arises
+   * @static
+   * @param {*} userId
+   * @param {*} roleId
+   * @returns {*} admin
+   * @memberof UserRoleManager
+   */
+  static async createAdminRole(userId, roleId) {
+    const userRole = await UserRoles.create({ userId, roleId });
+    return userRole;
+  }
+
+  /**
+   *
+   * @description the manager create admin role if the need arises
+   * @static
+   * @param {*} role
+   * @returns {*} admin
+   * @memberof UserRoleManager
+   */
+  static async createRole(role) {
+    const newRole = await Roles.create({ role });
+    return newRole;
+  }
+
+  /**
+   *
+   * @description The manager was created to check if role already exist in the role table
+   *  because role is meant to be unique
+   * @static
+   * @param {*} newRole
+   * @returns {*} admin
+   * @memberof UserRoleManager
+   */
+  static async findRole(newRole) {
+    const options = {
+      where: {
+        role: {
+          [Op.eq]: newRole
+        }
+      }
+    };
+    const role = await Roles.findOne(options);
+    return role;
+  }
+}
```

Comment:
Role creation and finding a role should be in the role model manager class.

Reply:
I have done it